openapi.yaml

```diff
@@ -0,0 +1,730 @@
+openapi: 3.0.3
+info:
+  title: IO Services CMS
+  description: |-
+    _This is the IO Service CMS based on the OpenAPI 3.0 specification._
+
+    ## Introduction
+    _TODO: write an introduction_
+
+    ### Some useful links:
+    - [IO-SERVICES-CMS Github codebase](https://github.com/pagopa/io-services-cms)
+    - [IO-SERVICES-CMS Source API definition ](https://github.com/pagopa/io-services-cms/openapi.yaml)
+
+  contact:
+    name: PagoPA S.p.A.
+    url: https://docs.pagopa.it/io-guida-tecnica/
+  version: 0.2.0
+servers:
+  - url: https://todo-url
+tags:
+  - name: services
+    description: Services API specification
+  - name: service-actions
+    description: Service Actions API specification
+  - name: service-release
+    description: Service Release API specification
+  - name: services-authorization
+    description: Services Authorization API specification
+  - name: organizations
+    description: Organizations API specification
+paths:
+  /services:
+    post:
+      tags:
+        - services
+      summary: Create a new service
+      description: |
+        Create a new Service with the attributes provided in the request payload.
+      operationId: createService
+      requestBody:
+        description: |
+          A service body payload
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServicePayload'
+        required: true
+      responses:
+        '201':
+          description: Service created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '400':
+          description: Invalid payload
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error
+      security:
+        - api_key: []
+    get:
+      tags:
+        - services
+      summary: Retrieve all services
+      description: Retrieve all services owned by the calling user
+      operationId: getServices
+      parameters: 
+        - name: limit
+          in: query
+          description: Number of services per page
+          required: true
+          schema:
+            type: number
+        - name: offset
+          in: query
+          description: Offset services number
+          required: false
+          schema:
+            type: number
```

Comment:
It depends, because if we retrieve services starting from subscriptions, Azure Apim has `$top` and `$skip` parameters and returns total `count` of retrieved subscriptions.

Reply:
Let's reason agnostically: what are the common pagination patterns? I see:
1. `offset` and `limit`
2. `page` and `page_size`
3. `next_id` (provide the id of the next document to be retrieved)

We can usually make `1` out of `2` and vice versa. `3` is tricky as it's often very bound to the database query language.

Reply:
2e38485e02312f20ce473591fc854b1e11de2913